test/test_jit_llga_fuser.py

```diff
@@ -366,6 +366,23 @@ def forward_cat(*inputs):
                 _, graph = self.checkTrace(cat_along_dim(d), [x, x, x], dtype)
                 self.assertGraphContainsExactly(graph, LLGA_FUSION_GROUP, 1)
 
+    @onlyCPU
+    def test_cat_check_type_promotion(self):
+        def cat_along_dim(d):
+            def forward_cat(*inputs):
+                return torch.cat(inputs, d)
+            return forward_cat
+
+        for xshape, yshape, dim in [
+            [[4, 8, 8, 16], [4, 8, 8, 24], 3],
+            [[64, 8, 32], [64, 24, 32], 1],
+            [[2048, 64], [1024, 64], 0],
+        ]:
+            x = torch.rand(xshape)
+            y = torch.rand(yshape, dtype=torch.bfloat16)
```

Comment:
This won't happen if users do autocast, right? I guess autocast would promote types for `cat`.

Reply:
Oh, maybe that's why we didn't need this commit 25 days back. A commit changed some AMP functionality 22 days back. I'll dig deeper.

Reply:
I had thought we didn't support AMP for `cat`, so I opened this PR.

`cat`'s AMP implementation seems to be broken for eager mode on CPU. So, I'll close this PR, as it's redundant. Thanks!